hack/generator/pkg/astmodel/object_type.go

```diff
@@ -212,6 +304,15 @@ func (objectType *ObjectType) WithProperties(properties ...*PropertyDefinition)
 	return result
 }
 
+// WithoutProperty creates a new ObjectType without the specified field
+func (objectType *ObjectType) WithoutProperty(name PropertyName) *ObjectType {
+	// Create a copy of objectType to preserve immutability
+	result := objectType.copy()
```

Comment:
We should be consistent with the copy technique used:

``` go
	// Create a copy of objectType to preserve immutability
	result := objectType.copy()
```
vs
```
	// Create a copy of objectType to preserve immutability
	result := *objectType
```

I'd favour the `.copy()` version purely for explicitness, though it does lessen the amount of memory sharing between instances. Not a big issue.

Reply:
Ah, this is you getting at what I was getting at above basically -- some places we do `foo := *foo` and others we have our own `copy`. The `foo := *foo` pattern threw me because at first I was like: "Wait that's not a strong enough copy" and then I realized that down below we were allocating a new collection and modifying that newly allocated collection.

I'm slightly in favor of `copy` too but do admit it makes more work for us (have to write it), _and_ increases the number of allocations... but it does seem safer (less likely to somehow result in mutating an object meant to be immutable) to me.